lib/oli_web/live/workspaces/course_author/editor_live.ex

```diff
@@ -29,86 +30,93 @@ defmodule OliWeb.Workspaces.CourseAuthor.Curriculum.EditorLive do
       </div>
-    <% end %>
-
-    <div id="editor" style="width: 95%;" class="container mx-auto">
-      <%= React.component(@ctx, "Components.PageEditor", @raw_context, id: "page_editor") %>
-    </div>
 
-    <div class="container mx-auto mt-5">
-      <%= live_render(@socket, OliWeb.CollaborationLive.CollabSpaceConfigView,
-        id: "collab-space-#{@project_slug}-#{@revision_slug}",
-        session: %{
-          "collab_space_config" => @collab_space_config,
-          "project_slug" => @project_slug,
-          "resource_slug" => @revision_slug
-        }
-      ) %>
-    </div>
+      <div class="container mx-auto mt-5">
+        <%= live_render(@socket, OliWeb.CollaborationLive.CollabSpaceConfigView,
+          id: "collab-space-#{@project_slug}-#{@revision_slug}",
+          session: %{
+            "collab_space_config" => @collab_space_config,
+            "project_slug" => @project_slug,
+            "resource_slug" => @revision_slug
+          }
+        ) %>
+      </div>
 
-    <%= render_prev_next_nav(assigns) %>
+      <%= render_prev_next_nav(assigns) %>
+    </.scripts_wrapper>
     """
   end
 
-  defp render_advanced(assigns) do
-    ~H"""
-    <script type="text/javascript" src={Routes.static_path(OliWeb.Endpoint, "/js/authoring.js")}>
-    </script>
-
-    <%= for %{slug: slug, authoring_script: script} <- @activity_types do %>
-      <%= if slug == "oli_adaptive" do %>
-        <script type="text/javascript" src={Routes.static_path(OliWeb.Endpoint, "/js/#{script}")}>
-        </script>
-      <% end %>
-    <% end %>
+  @impl true
+  def handle_event("survey_scripts_loaded", %{"error" => _}, socket) do
+    {:noreply, assign(socket, error: true, maybe_scripts_loaded: true)}
+  end
 
-    <%= for script <- @part_scripts do %>
-      <script type="text/javascript" src={Routes.static_path(OliWeb.Endpoint, "/js/#{script}")}>
-      </script>
-    <% end %>
+  def handle_event("survey_scripts_loaded", _params, socket) do
+    {:noreply, assign(socket, maybe_scripts_loaded: true)}
+  end
 
-    <div id="editor" class="container">
-      <%= React.component(@ctx, "Components.Authoring", @app_params, id: "authoring_editor") %>
+  defp maybe_show_error(assigns) do
+    ~H"""
+    <div :if={@error} class="alert alert-danger m-0 flex flex-row justify-between w-full" role="alert">
+      Something went wrong when loading the JS dependencies.
     </div>
+    """
+  end
 
-    <%= render_prev_next_nav(assigns) %>
+  slot :inner_block, required: true
+  attr(:socket, :any, required: true)
+  attr(:error, :boolean, required: true)
+  attr(:maybe_scripts_loaded, :boolean, required: true)
+
+  def scripts_wrapper(assigns) do
+    ~H"""
+    <div id="eventIntercept" phx-hook="LoadSurveyScripts">
+      <%= if connected?(@socket) and @maybe_scripts_loaded do %>
+        <.maybe_show_error error={@error} />
+        <%= render_slot(@inner_block) %>
+      <% else %>
+        <.loader />
+      <% end %>
+    </div>
     """
   end
```

Comment:
In a future iteration, this can be extracted into a generic module to be used whenever we need to load JavaScript dependencies via a hook.